service/controller/clusterapi/v30/cloudconfig/tccp.go

```diff
@@ -17,111 +17,117 @@ import (
-			RandomKeyTmplSet: randomKeyTmplSet,
+			awsConfigSpec: cmaClusterToG8sConfig(t.config, cr, labels),
+			baseExtension: baseExtension{
+				cluster:       cr,
+				encrypter:     t.config.Encrypter,
+				encryptionKey: cc.Status.TenantCluster.Encryption.Key,
+			},
+			cc:               cc,
+			clusterCerts:     clusterCerts,
+			randomKeyTmplSet: randomKeyTmplSet,
 		}
-		params.Hyperkube.Apiserver.Pod.CommandExtraArgs = c.k8sAPIExtraArgs
-		params.Hyperkube.Kubelet.Docker.CommandExtraArgs = c.k8sKubeletExtraArgs
-		params.ImagePullProgressDeadline = c.imagePullProgressDeadline
-		params.RegistryDomain = c.registryDomain
-		params.SSOPublicKey = c.ssoPublicKey
+		params.Hyperkube.Apiserver.Pod.CommandExtraArgs = t.config.APIExtraArgs
+		params.Hyperkube.Kubelet.Docker.CommandExtraArgs = t.config.KubeletExtraArgs
+		params.ImagePullProgressDeadline = t.config.ImagePullProgressDeadline
+		params.RegistryDomain = t.config.RegistryDomain
+		params.SSOPublicKey = t.config.SSOPublicKey
 
-		ignitionPath := k8scloudconfig.GetIgnitionPath(c.ignitionPath)
+		ignitionPath := k8scloudconfig.GetIgnitionPath(t.config.IgnitionPath)
 		params.Files, err = k8scloudconfig.RenderFiles(ignitionPath, params)
 		if err != nil {
-			return "", microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
 	}
 
-	var newCloudConfig *k8scloudconfig.CloudConfig
+	var templateBody []byte
 	{
-		cloudConfigConfig := k8scloudconfig.DefaultCloudConfigConfig()
-		cloudConfigConfig.Params = params
-		cloudConfigConfig.Template = k8scloudconfig.MasterTemplate
+		c := k8scloudconfig.CloudConfigConfig{
+			Params:   params,
+			Template: k8scloudconfig.MasterTemplate,
+		}
 
-		newCloudConfig, err = k8scloudconfig.NewCloudConfig(cloudConfigConfig)
+		cloudConfig, err := k8scloudconfig.NewCloudConfig(c)
 		if err != nil {
-			return "", microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
 
-		err = newCloudConfig.ExecuteTemplate()
+		err = cloudConfig.ExecuteTemplate()
 		if err != nil {
-			return "", microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
-	}
 
-	return newCloudConfig.String(), nil
-}
+		templateBody = []byte(cloudConfig.String())
+	}
 
-// RandomKeyTmplSet holds a collection of rendered templates for random key
-// encryption via KMS.
-type RandomKeyTmplSet struct {
-	APIServerEncryptionKey string
+	return templateBody, nil
 }
 
 type MasterExtension struct {
 	awsConfigSpec g8sv1alpha1.AWSConfigSpec
 	baseExtension
-
 	// TODO Pass context to k8scloudconfig rendering fucntions
 	//
 	//	See https://github.com/giantswarm/giantswarm/issues/4329.
 	//
-	ctlCtx *controllercontext.Context
-
-	ClusterCerts     certs.Cluster
-	RandomKeyTmplSet RandomKeyTmplSet
+	cc               *controllercontext.Context
+	clusterCerts     certs.Cluster
+	randomKeyTmplSet RandomKeyTmplSet
 }
 
 func (e *MasterExtension) Files() ([]k8scloudconfig.FileAsset, error) {
-	// TODO Pass context to k8scloudconfig rendering functions.
```

Comment:
This shouldn't be removed yet - my change in `k8scc` which I will pick up later passes `ctx` through to `Files()`.

Reply:
It is still there. I just removed the duplicates that appeared useless to me. IMO it is ok to have it once in the file. We will get to make this better soon with what you started to push recently.

Reply:
:+1: Okay, didn't catch the duplicates.